sycl/source/detail/graph_impl.cpp

```diff
@@ -201,6 +183,7 @@ graph_impl::add(const std::vector<std::shared_ptr<node_impl>> &Dep) {
   // Add any deps from the vector of extra dependencies
   Deps.insert(Deps.end(), MExtraDependencies.begin(), MExtraDependencies.end());
 
+  MNodeStorage.push_back(NodeImpl);
   // TODO: Encapsulate in separate function to avoid duplication
```

Comment:
Could I be cheeky and ask for this TODO to be actioned in this PR?

Reply:
I've added a new function `addDepsToNode()` for this!